### ISSUE TITLE
Benchmark of 0.13 Release

### ISSUE THREAD

**Issue by erylee** [emqtt/emqttd#373]
Long schedule issue: 
```
13:32:14.665 <0.237.0> [error] long_schedule warning: pid = <0.142.0>, info: [{timeout,233},
                                               {in,{mnesia_locker,loop,1}},
                                               {out,{mnesia_locker,loop,1}}]
[{initial_call,{proc_lib,init_p,5}},{current_function,{mnesia_locker,loop,1}},{registered_name,mnesia_locker},{status,waiting},{message_queue_len,0},{group_leader,<0.135.0>},{priority,normal},{trap_exit,true},{reductions,7536808},{last_calls,false},{catchlevel,2},{trace,0},{suspending,[]},{sequential_trace_token,[]},{error_handler,error_handler},{memory,29528},{total_heap_size,3573},{heap_size,2586},{stack_size,15},{min_heap_size,233}]
13:32:14.668 <0.237.0> [error] long_schedule warning: pid = <0.144.0>, info: [{timeout,233},
                                               {in,{mnesia_tm,doit_loop,1}},
                                               {out,{mnesia_tm,doit_loop,1}}]
```

Mnesia info:

```
# ./bin/emqttd_ctl mnesia
===> System info in version "4.13.1", debug level = none <===
opt_disc. Directory "/opt/emqttd-0.13.0/data/mnesia/emqttd@127.0.0.1" is used.
use fallback at restart = false
running db nodes   = ['emqttd@127.0.0.1']
stopped db nodes   = []
master node tables = []
remote             = []
ram_copies         = [queue,retained,session,subscriber,topic,trie,trie_node]
disc_copies        = [mqtt_admin,schema]
disc_only_copies   = []
[{'emqttd@127.0.0.1',disc_copies}] = [mqtt_admin,schema]
[{'emqttd@127.0.0.1',ram_copies}] = [subscriber,trie_node,session,topic,
                                     retained,queue,trie]
432553 transactions committed, 9 aborted, 87427 restarted, 0 logged to disc
0 held locks, 0 in queue; 0 local transactions, 0 remote
0 transactions waits for other nodes: []
```

**Comment by erylee** [comment]
[benchmark for 0.13.0 release](https://github.com/emqtt/emqttd/wiki/benchmark-for-0.13.0-release)

**Comment by erylee** [comment]
# ./bin/emqttd_ctl recon bin_leak

```
{<10106.187.0>,-181,
 [emqttd_pubsub_1,
  {current_function,{gen_server2,process_next_msg,1}},
  {initial_call,{proc_lib,init_p,5}}]}
{<10106.194.0>,-63,
 [emqttd_pubsub_8,
  {current_function,{gen_server2,process_next_msg,1}},
  {initial_call,{proc_lib,init_p,5}}]}
{<10106.189.0>,-59,
 [emqttd_pubsub_3,
  {current_function,{gen_server2,process_next_msg,1}},
  {initial_call,{proc_lib,init_p,5}}]}
{<10106.228.0>,-58,
 [emqttd_alarm,
  {current_function,{gen_event,fetch_msg,5}},
  {initial_call,{proc_lib,init_p,5}}]}
{<10106.192.0>,-57,
 [emqttd_pubsub_6,
  {current_function,{gen_server2,process_next_msg,1}},
  {initial_call,{proc_lib,init_p,5}}]}
{<10106.227.0>,-5,
 [emqttd_broker,
  {current_function,{gen_server,loop,6}},
  {initial_call,{proc_lib,init_p,5}}]}
{<10106.235.0>,-4,
 [emqttd_sysmon,
  {current_function,{gen_server,loop,6}},
  {initial_call,{proc_lib,init_p,5}}]}

...
```

**Comment by erylee** [comment]
Fixed in 0.14. Hibernate if subscriber 'DOWN' received to gc binary:

```
handle_info({'DOWN', _Mon, _Type, DownPid, _Info}, State) ->

    Routes = ?ROUTER:lookup_routes(DownPid),

    %% Delete all routes of the process
    ?ROUTER:delete_routes(DownPid),

    ?HELPER:aging([Topic || Topic <- Routes, not ?ROUTER:has_route(Topic)]),

    {noreply, State, hibernate};
```
